HPXMLtoOpenStudio/resources/output.rb

```diff
@@ -172,6 +173,107 @@ def self.apply_unmet_hours_ems_program(model, hpxml_osm_map, hpxml_header)
     return season_day_nums
   end
 
+  # Apply EMS program to adjust discharge power based on ambient temperature.
+  # An EMS program models the effect of ambient temperature on the effective power output, scales power with the fraction charged at home, and calculates the unmet driving hours.
+  #
+  # @param model [OpenStudio::Model::Model] OpenStudio Model object
+  # @param hpxml_osm_map [Hash] Map of HPXML::Building objects => OpenStudio Model objects for each dwelling unit
+  # @return [nil]
+  def self.apply_unmet_driving_hours_ems_program(model, hpxml_osm_map)
+    return if hpxml_osm_map.keys.map { |hpxml_bldg| hpxml_bldg.vehicles.size }.sum == 0
+
+    temp_sensor = Model.add_ems_sensor(
+      model,
+      name: 'site_temp',
+      output_var_or_meter_name: 'Site Outdoor Air Drybulb Temperature',
+      key_name: 'Environment'
+    )
+
+    # Power adjustment vs ambient temperature curve; derived from most recent data in Figure 9 of https://www.nrel.gov/docs/fy23osti/83916.pdf
+    # This adjustment scales power demand based on ambient temperature, and encompasses losses due to battery and space conditioning (i.e., discharging losses), as well as charging losses.
+    coefs = [1.412768, -3.910397E-02, 9.408235E-04, 8.971560E-06, -7.699244E-07, 1.265614E-08]
+    power_curve = ''
+    coefs.each_with_index do |coef, i|
+      power_curve += "+(#{coef}*(site_temp_adj^#{i}))"
+    end
+    power_curve = power_curve[1..]
+
+    # EMS program
+    driving_hrs = 'unmet_driving_hours'
+    unit_driving_hrs = 'unit_driving_unmet_hours'
+    ev_discharge_program = Model.add_ems_program(
+      model,
+      name: 'ev_discharge_program'
+    )
+    ev_discharge_program.additionalProperties.setFeature('ObjectType', Constants::ObjectTypeBEVDischargeProgram)
+    ev_discharge_program.addLine("Set #{driving_hrs} = 0")
+    ev_discharge_program.addLine("  Set site_temp_adj = #{temp_sensor.name}")
+    ev_discharge_program.addLine("  If #{temp_sensor.name} < #{UnitConversions.convert(0, 'F', 'C').round(3)}")
+    ev_discharge_program.addLine("    Set site_temp_adj = #{UnitConversions.convert(0, 'F', 'C').round(3)}")
+    ev_discharge_program.addLine("  ElseIf #{temp_sensor.name} > #{UnitConversions.convert(100, 'F', 'C').round(3)}")
+    ev_discharge_program.addLine("    Set site_temp_adj = #{UnitConversions.convert(100, 'F', 'C').round(3)}")
+    ev_discharge_program.addLine('  EndIf')
+    ev_discharge_program.addLine("  Set power_mult = #{power_curve}")
```

Comment:
This line was previously before "Set site_temp_adj = ...", which I believe was a bug.